### PR TITLE
Port to cmake + android studio: Ready to merge

### PR DESCRIPTION
**Pull request by ggfan**
Seems ok for integration, please have a look when you have time. Surprisingly that shader compiling is working in android studio 2.2.0-beta1: if shader has no errors, it will compile through!

### CODE REVIEW

File: tutorial03_traceable_layers/app/src/main/jni/Application.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+APP_ABI := armeabi-v7a arm64-v8a x86 x86_64 mips mips64
+APP_PLATFORM := android-22
+APP_STL := gnustl_static
+APP_CPPFLAGS += -std=c++11 -DVK_PROTOTYPES -Wall -Werror -Wno-unused-variable -Wno-unused-function -Wno-unused-const-variable -DUSE_DEBUG_EXTENTIONS -DVK_USE_PLATFORM_ANDROID_KHR
```
Comment (courtney-g) [inline tutorial03_traceable_layers/app/src/main/jni/Application.mk]: This should probably move to Android.mk. Otherwise these will apply to any modules being imported. That was an issue for me when trying to include validation layers and the compile options don't match what they expected.

File: tutorial03_traceable_layers/layerlib/proguard-rules.pro [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# directive in build.gradle.
+#
+# For more details, see
+#   http://developer.android.com/guide/developing/tools/proguard.html
+
+# Add any project specific keep options here:
+
+# If your project uses WebView with JS, uncomment the following
+# and specify the fully qualified class name to the JavaScript interface
+# class:
```
Comment (courtney-g) [inline tutorial03_traceable_layers/layerlib/proguard-rules.pro]: Is this file necessary?
Seems like it may be system specific.

File: tutorial05_triangle/settings.gradle [hunk 3]
```diff
@@ -1,2 +1 @@
-include ':app', ':nativeactivity'
-project(':nativeactivity').projectDir = new File(settingsDir, '../common/nativeactivity')
+include ':app'
```
Comment (courtney-g) [inline tutorial05_triangle/settings.gradle]: Missing newline?

File: tutorial03_traceable_layers/layerlib/proguard-rules.pro [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Add project specific ProGuard rules here.
+# By default, the flags in this file are appended to flags specified
+# in /Users/gfan/dev/sdk_current/tools/proguard/proguard-android.txt
```
Comment (courtney-g) [inline tutorial03_traceable_layers/layerlib/proguard-rules.pro]: Probably shouldn't have system specific path in upstream bits.

File: tutorial03_traceable_layers/layerlib/build.gradle [hunk 5]
```diff
@@ -0,0 +1,65 @@
+apply plugin: 'com.android.library'
+
+// Both cmake and ndkbuild are simple, pick one and feel free to remove the
+// other one. This code is tested again Android Studio 2.2 Beta + NDK-12
```
Comment (courtney-g) [inline tutorial03_traceable_layers/layerlib/build.gradle]: again => against

File: tutorial03_traceable_layers/layerlib/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+add_library(VkLayer_object_tracker SHARED
+            ${LAYER_DIR}/layer-src/object_tracker/object_tracker.cpp
+            ${LAYER_DIR}/common/vk_layer_table.cpp)
+target_link_libraries(VkLayer_object_tracker log android layer_utils)
+
+add_library(VkLayer_threading SHARED
+            ${LAYER_DIR}/layer-src/threading/threading.cpp
+            ${LAYER_DIR}/common/vk_layer_table.cpp)
+target_link_libraries(VkLayer_threading log android layer_utils)
```
Comment (courtney-g) [inline tutorial03_traceable_layers/layerlib/CMakeLists.txt]: It's not clear to me what happens to these libraries. Are they linked with the app? Or just included in the app's libs folder?

File: tutorial03_traceable_layers/app/src/main/jni/Application.mk (hunk 1 above)
Comment (ggfan) [inline tutorial03_traceable_layers/app/src/main/jni/Application.mk]: yeah, I will do that.

File: tutorial03_traceable_layers/layerlib/proguard-rules.pro (hunk 2 above)
Comment (ggfan) [inline tutorial03_traceable_layers/layerlib/proguard-rules.pro]: this was taken from template when "new" a java app for android, not much dig into it, might just leave it there: it is not native related.  for future java guys to add new stuff in there.

File: tutorial05_triangle/settings.gradle (hunk 3 above)
Comment (ggfan) [inline tutorial05_triangle/settings.gradle]: I need delete the nativeactivity actually, good catch

File: tutorial03_traceable_layers/layerlib/proguard-rules.pro (hunk 4 above)
Comment (ggfan) [inline tutorial03_traceable_layers/layerlib/proguard-rules.pro]: fixed: I did not see the put the sdk real path there.

File: tutorial03_traceable_layers/layerlib/build.gradle (hunk 5 above)
Comment (ggfan) [inline tutorial03_traceable_layers/layerlib/build.gradle]: Ha, fixed, thanks!

File: tutorial03_traceable_layers/layerlib/CMakeLists.txt (hunk 6 above)
Comment (ggfan) [inline tutorial03_traceable_layers/layerlib/CMakeLists.txt]: they are built here, and linked into the libs folder in upper layer build.gradle
            main {
                dependencies {
                    project ':layerlib'
                }
            }

File: tutorial03_traceable_layers/layerlib/proguard-rules.pro (hunk 2 above)
Comment (courtney-g) [inline tutorial03_traceable_layers/layerlib/proguard-rules.pro]: Is it necessary?
Will things build / run without this file?

**Comment by courtney-g** [comment]
LGTM
